### PR TITLE
Add a simple generalized autobumper for Prow.

### PR DESCRIPTION
**Pull request by cjwagner**
This is a quick and easy generalized Prow autobumper that uses updated versions of the old `autobump.sh` and `bump.sh` scripts.
We should replace this will a proper Go tool eventually, but I know that @stevekuznetsov wants to revamp the git publishing client first and @fejta also pointed out that reusing the shell scripts will get us something usable faster than writing the tool in Go. So this is far from polished, but it should get the job done for the time being.

/assign @fejta @stevekuznetsov @clarketm

### CODE REVIEW

File: prow/cmd/autobump/BUILD.bazel [hunk 1]
```diff
@@ -0,0 +1,29 @@
+load("@io_bazel_rules_docker//container:image.bzl", "container_image")
+
+container_image(
+    name = "image",
+    base = "@gcloud-go//image",
```
Comment (fejta) [inline prow/cmd/autobump/BUILD.bazel]: Does this image have tac?

File: prow/cmd/autobump/autobump.sh [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+set -o nounset
+set -o pipefail
+
+PLANK_DEPLOYMENT_FILE="${PLANK_DEPLOYMENT_FILE:-}"
+GH_ORG="${GH_ORG:-}"
+GH_REPO="${GH_REPO:-}"
+FORK_GH_REPO="${FORK_GH_REPO:-${GH_REPO}}"
+
+
+# TODO(fejta): rewrite this in a better language REAL SOON  <-lol
```
Comment (fejta) [inline prow/cmd/autobump/autobump.sh]: This script is undead

File: prow/cmd/autobump/autobump.sh [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+	title="Bump prow from ${old_version} to ${version}"
+	git add -u
+	git commit -m "${title}"
+	echo -e "Pushing commit to ${user}/${FORK_GH_REPO}:autobump..." >&2
+	git push -f "https://${user}:$(cat "${token}")@github.com/${user}/${FORK_GH_REPO}" HEAD:autobump
+
+	echo "Creating PR to merge ${user}:autobump into master..." >&2
+	comparison=$(extract-commit "${old_version}")...$(extract-commit "${version}")
+	/pr-creator \
```
Comment (fejta) [inline prow/cmd/autobump/autobump.sh]: `//robots/pr-creator:pr-creator` becomes `/pr-creator`? That seems surprising.

File: prow/cmd/autobump/autobump.sh [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+	  --title="${title}" --match-title="Bump prow from" \
+	  --body="Included changes: https://github.com/kubernetes/test-infra/compare/${comparison}" \
+	  --source="${user}":autobump \
+	  --confirm
+
+	echo "autobump.sh completed successfully!" >&2
+}
+
+user-from-token() {
+	user=$(curl -H "Authorization: token $(cat ${token})" "https://api.github.com/user" | sed -n "s/\s\+\"login\": \"\(.*\)\",/\1/p")
```
Comment (stevekuznetsov) [inline prow/cmd/autobump/autobump.sh]: This might leak on stderr

Comment (fejta) [inline prow/cmd/autobump/autobump.sh]: `$(cat "$token")`

File: prow/cmd/autobump/BUILD.bazel (hunk 1 above)
Comment (cjwagner) [reply prow/cmd/autobump/BUILD.bazel]: Surprisingly, yes.

File: prow/cmd/autobump/autobump.sh (hunk 4 above)
Comment (cjwagner) [reply prow/cmd/autobump/autobump.sh]: Updated to send stderr to `/dev/null`.

File: prow/cmd/autobump/autobump.sh (hunk 3 above)
Comment (cjwagner) [reply prow/cmd/autobump/autobump.sh]: That is how bazel handles the file dependency :man_shrugging:

File: prow/cmd/autobump/autobump.sh (hunk 4 above)
Comment (cjwagner) [reply prow/cmd/autobump/autobump.sh]: Added the quotes, but I'm assuming I should keep the curly braces.

**Comment by cjwagner** [comment]
> Wait. why did we need this in addition to the go tool?

See the PR description. This is just a quick and easy way to get the automation we want working today. We can replace this with the go tool when it is ready.
/hold cancel

**Pull request merged**